src/core/xref.js

```diff
@@ -0,0 +1,134 @@
+// Automatically adds external refernces
+// https://github.com/w3c/respec/issues/1662
+
+import { norm } from "core/utils";
+import { pub } from "core/pubsubhub";
+
+/**
+ * main external reference driver
+ * @param {Object} conf respecConfig
+ * @param {Array:Elements} elems possibleExternalLinks
+ */
+export async function run(conf, elems) {
+  const xrefMap = createXrefMap(elems);
+
+  const query = createXrefQuery(xrefMap);
+  const results = await fetchXrefs(query, conf.xref.url);
+
+  addDataCiteToTerms(results, xrefMap, conf);
+}
+
+/**
+ * maps term to elements and context
+ * @param {Array:Elements} elems
+ * @returns {Map} term => [ { elem } ]
+ */
+function createXrefMap(elems) {
+  return elems.reduce((map, elem) => {
+    const term = norm(elem.textContent);
+    const xrefsForTerm = map.has(term) ? map.get(term) : [];
+    xrefsForTerm.push({ elem });
+    return map.set(term, xrefsForTerm);
+  }, new Map());
+}
+
+/**
+ * creates a body for POST request to API
+ * @param {Map} xrefs
+ * @returns {Object} { keys: [{ term }] }
+ */
+function createXrefQuery(xrefs) {
+  const queryKeys = [...xrefs.keys()].reduce((queryKeys, term) => {
+    return queryKeys.add(JSON.stringify({ term }));
+  }, new Set());
+  return { keys: [...queryKeys].map(JSON.parse) };
+}
+
+// fetch from network
+async function fetchXrefs(query, url) {
+  return await simulateShepherd(query, url);
```

Comment:
I like the idea of arbitrary end points that happen to respond with our preferred data structure. 

So here, we can build the actual thing without needing to say "simulateShepherd"... because we are doing the real thing :)